### DISCUSSION TITLE
[Suggestions] Game, exchange the only creature in the army for another creature by dragging it onto the desired creature's troop

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#5469] [Ideas]
I suggest to add an improvement that allows to exchange the only creature in the army for another creature by dragging it onto the desired creature's troop.

First example, drag and drop the halfling on the troop of boars so that one of the boars replaces the only halfling of the hero's army.

In fheroes2:

![Exchange the one creature for another flèche 001](https://user-images.githubusercontent.com/43583994/170660096-6bba69df-6ece-4873-8c67-3fe4e36c9698.png)

Second example, drag and drop the sprite on the troop of boars so that one of the boars replaces the only sprite of the hero's army.

In fheroes2:

![Exchange the one creature for another flèche 002](https://user-images.githubusercontent.com/43583994/170660163-3f1f9867-dd5e-4219-84d9-1dc820f6d312.png)

**Comment by Branikolog** [comment]
@LeHerosInconnu 
So as a result of both these examples, heroes with 1 starting creature would have a stack of 3 boars. Am I right?

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> @LeHerosInconnu So as a result of both these examples, heroes with 1 starting creature would have a stack of 3 boars. Am I right?

No. :)
In the first example, there will be a troop of 7 halflings and a troop of 2 boars in garrison, and the hero will have only one boar in his army.
In the second example, the sprite will be transferred to the free slot of the first hero's army, and the second hero will have only one boar in his army.
Of course, if there is no room to perform the action, nothing happens.

**Reply by zenseii** [reply]
While I can understand why you would want to do this I believe the standard action from a left-button drag and drop would be to exchange 1 halfling with 3 boars.

To acheive what @LeHerosInconnu wants you could use the mouse wheel button for mouse wheel only users (right click is for info). For keyboard and mouse this would be similar to the "chipping off" of one single creature that you get when holding ctrl+ and left click except with this you also perform a swap of the creatures.

I have to note that personally I would not leave an army with only 1 stack of 1 unit in Heroes of Surrender or Retreat II. My rule for such gatherer heroes with small armies is to in this case have at least 2 stacks of boars with 1 unit, preferably 3 stacks spread out. Yes, you could always meet an enemy hero with 4-5 stacks of phoenixes or just one phoenix/champion plus chain lightning, but the chances are generally smaller for that. The 3rd boar might even allow you to sneak in a second damage spell like lightning bolt before you sayonara back home to your castle.

**Reply by LeHerosInconnu** [reply]
@zenseii.

> While I can understand why you would want to do this I believe the standard action from a left-button drag and drop would be to exchange 1 halfling with 3 boars.

But this can already be done with first a left click on the troop of one halfling and then a left click on the troop of three boars.
Using the drag and drop for this is therefore not necessary.
Why "waste" the possibility of another action with the drag and drop?

> I have to note that personally I would not leave an army with only 1 stack of 1 unit in Heroes of Surrender or Retreat II. My rule for such gatherer heroes with small armies is to in this case have at least 2 stacks of boars with 1 unit, preferably 3 stacks spread out. Yes, you could always meet an enemy hero with 4-5 stacks of phoenixes or just one phoenix/champion plus chain lightning, but the chances are generally smaller for that. The 3rd boar might even allow you to sneak in a second damage spell like lightning bolt before you sayonara back home to your castle.

It also depends on the hero's location on the adventure map.
For example, if he is in the player's territory with very little risk of being attacked.
I myself use different configurations for these secondary heroes, sometimes with one fast troop (example: one boar), sometimes with two or three fast troops (example: two or three troops of one boar), or with five fast troops (example: five troops of one boar).

**Reply by zenseii** [reply]
> But this can already be done with first a left click on the troop of one halfling and then a left click on the troop of three boars. Using the drag and drop for this is therefore not necessary. Why "waste" the possibility of another action with the drag and drop?


Yes, you're definitely right. I think the underlying issue here is that we currently have ways of input that could be perceived as doing the same thing and this will in the future be something we have to solve. The only important thing right now is that we have the code implemented for these commands so that we easily can change what they do in the code later on. In this regard I'm fine with implementing what you suggested above and if we realize at a later point that there was a much more logical thing to do with drag and drop in this situation then we can simply change it.

For these changes where what choice to make is not clear, I think the best course of action is to just try one of them instead of discussing it ad infinitum. Some things can be implemented perfectly at first, while others just need real play-testing.
